docs/USAGE-GUIDE.md

```diff
@@ -500,6 +500,208 @@ legacy SHA-1 algorithms in CertificateVerify messages if TLS1.2 has been negotia
 |   "20200207"   |      X       |      X     |   X    |
 |   "rfc9151"    |              |      X     |        |
 
+## Sending and Receiving
+
+### Basic IO setup
+
+By default, s2n-tls sends and receives data using a provided file descriptor
+(usually a socket) and the read/write system calls. The file descriptor can be set with
+`s2n_connection_set_fd()`, or separate read and write file descriptors can be
+set with `s2n_connection_set_read_fd()` and `s2n_connection_set_write_fd()`.
+The file descriptor set should be active and connected.
+
+In general the application is free to configure the file descriptor as preferred,
+including socket options. s2n-tls itself sets a few socket options:
+* If available, TCP_QUICKACK is used during the TLS handshake
+* If available and enabled via `s2n_connection_use_corked_io()`, TCP_CORK is
```

Comment:
Do we need a bigger explanation about corked IO? Since people have been asking about how we deal with it :)

Reply:
I've only noticed one customer asking for one very specific use case. I don't necessarily want to over-index on the latest question :/

But if you think it's necessary, i can add a separate section for corking.